Network Hub - Requested & Received API enhancements

tarentomaheshvakkund opened the pull request:
1. First the redis key would be constructed -for requested/received and count APIS.
2. Fetch the details from the redis .
3. If there is no data in the redis cache - then fetch the data from the database and append it to the redis - even if it is empty or 0.
4. The next db call will only happen after the ttl is expired.

tarentomaheshvakkund committed:
Network Hub - Requested & Received API enhancements

1. First the redis key would be constructed -for requested/received and count APIS.
2. Fetch the details from the redis .
3. If there is no data in the redis cache - then fetch the data from the database and append it to the redis - even if it is empty or 0.
4. The next db call will only happen after the ttl is expired.

karthik-tarento commented on src/main/java/org/sunbird/cb/hubservices/serviceimpl/ConnectionService.java:
```diff
@@ -298,64 +298,63 @@ public Response findAllConnectionsIdsByStatusV2(String userId, String status, in
+			boolean nodesCached = StringUtils.isNotEmpty(cachedNodesJson);
+			boolean countCached = StringUtils.isNotEmpty(cachedCountJson);
+			//If both are cached, use cache
+			if (nodesCached && countCached) {
```
instead of boolean just use the string is not empty method directly..

tarentomaheshvakkund replied:
Made the code changes .

karthik-tarento commented on src/main/java/org/sunbird/cb/hubservices/serviceimpl/ConnectionService.java:
```diff
@@ -298,64 +298,63 @@ public Response findAllConnectionsIdsByStatusV2(String userId, String status, in
-				logger.info("No cached nodes found for user: " + userId + " in direction: " + direction);
-				nodes  = nodeService.getNodes(userId, relationProperties, direction, offset, limit, null);
+				//If cache miss, fetch from DB and cache the results
+				logger.info("Cache miss for userId: {} (direction: {}). Fetching from DB.", userId, direction);
```
if neo4j query returns empty -- how the if condition will work... ?
do we execute neo4j query again and again ?

tarentomaheshvakkund replied:
Only once the query will be executed if the data is not available it would be cached as empty data and count 0.
The db call will happen once the ttl is expired.

<img width="692" height="250" alt="image" src="https://github.com/user-attachments/assets/f8784f0e-5156-4c46-bc3e-323b2ccbb300" />

<img width="886" height="80" alt="image" src="https://github.com/user-attachments/assets/148eb174-91fa-4108-980a-a23dc74178b8" />

<img width="935" height="60" alt="image" src="https://github.com/user-attachments/assets/da105351-59c5-4fdf-9519-2fbed33e0488" />

tarentomaheshvakkund committed:
Code review changes

1. Using the StringUtils package directly.
2. Added conversion the String since i was getting casting exceptions.

tarentomaheshvakkund committed:
Formatted the code

tarentomaheshvakkund committed:
Code fix

1. We were convert to string two times removed redundant string conversion.

tarentomaheshvakkund committed:
Merge branch '4.8.27-dev-v6' into 4.8.27-dev-networkenhancements

tarentomaheshvakkund committed:
Merge branch '4.8.27-dev-v6' into 4.8.27-dev-networkenhancements

tarentomaheshvakkund committed:
Code fixes to enrich the user profile data
